pkg/infra/iac2/templates_compiler.go

```diff
@@ -430,6 +433,19 @@ func (tc TemplatesCompiler) handleIaCValue(v core.IaCValue) (string, error) {
 		default:
 			return "", errors.Errorf("unsupported resource type %T for '%s'", v.Resource, v.Property)
 		}
+	case resources.ALL_RESOURCES_ARN_IAC_VALUE:
+		method, ok := v.Resource.(*resources.ApiMethod)
+		if !ok {
+			return "", errors.Errorf("unsupported resource type %T for '%s'", v.Resource, v.Property)
+		}
+		verb := strings.ToUpper(method.HttpMethod)
+		if verb == "ANY" {
+			verb = "*"
+		}
+		accountId := resources.NewAccountId()
+		region := resources.NewRegion()
```

Comment:
Is there a reason the AccountId and the Region don't go on the struct if it depends on them?
Can this be derived from the RestApi arn instead? Something like
```ts
pulumi.interpolate`${api.arn}/*/%s${method.path}`
```

Non-blocking, but hopefully we don't leave it in for too long

Reply:
the rest api arn doesnt have either of them and doesnt follow the same format, i had tried that originally, but then had to redeploy our old version to see the correct format.

The permissions isnt dependent on the region or the account id necessarily so it would be weird to pack those in since they dont really fit on that or a method. Because region and account id are global i dont think this is the worst way to go about it for now, but i do agree theres room for improvement.

I dont think well see too many things like this,
just for reference what we want is 
`  "AWS:SourceArn": "arn:aws:execute-api:eu-west-1:338991950301:es3nxti7se/*/GET/read-file/{path}"`


and the api gateway arn is `arn:aws:api-gateway:eu-west-1::/rest-api/es3nxti7se` or something like that

Reply:
> The permissions isnt dependent on the region or the account id necessarily

It doesn't in a general sense, but when we render the IaC it definitely depends on it in code (ie, it would never be valid to put the Region or AccountId variables after this). The DAG is supposed to encode dependencies like this.

Reply:
yeah i mean we could put those in the core.IaCValue somehow so it has them or in the method so it would have them, but i feel like thats just muddying the water a little bit. The weird thing here is that were saying give me the arn to the method, but the iac doesnt have the ability to do that which is why we have to construct it via pulumi.

other iacs might have the ability to just grab the arn which is why to me the provider is doing what it needs to in saying the methods arn. Hard to know until we support more iacs though